Clinical-Genomics/scout
# Improved gene variants page (Search SNVs and INDELs)

This PR adds a functionality or fixes a bug.
- Better pagination footer on gene variants page (Search SNVs and INDELs)
- Display exact number of variants returned by the search
- Fix #1440 by refactoring the gene variants query
- Refactored and simplified code by removing `gene_variants` and `count_gene_variants` from the variant mongo adapter


<details>
<summary>Testing on cg-vm1 server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. Make sure the PR is pushed and available on [Docker Hub](https://hub.docker.com/repository/docker/clinicalgenomics/scout-server-stage)
1. `ssh <USER.NAME>@cg-vm1.scilifelab.se`
1. `sudo -iu hiseq.clinical`
1. `ssh localhost`
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `podman ps`
1. Stop the service with current deployed branch: `systemctl --user stop scout.target`
1. Start the scout service with the branch to test: `systemctl --user start scout@<this_branch>`
1. Make sure the branch is deployed: `systemctl --user status scout.target`
</details>

**How to test**:
1. Deploy latest release branch on scout-stage (cg-vm1)
2. As an admin, go to institute cust003 and look for gene variants (Search SNVs and INDELs) in POT1 with rank>12. It should be 51 (count them!)
3. Keep the browser function open and deploy this branch
4. In another browser window, go to the same page and check that it actually tells you the number of variants returned by the query.
5. The pagination should also be nicer
6. Modify your user as a non-admin user and make sure less variants are returned in this page

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [x] code approved by ML
